### PR TITLE
contributing: remove release_notes.md reference

### PR DESCRIPTION
**Pull request by tessr**
This file doesn't do anything.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
         with:
           go-version: '1.16'
 
-      - run: echo https://github.com/tendermint/tendermint/blob/${GITHUB_REF#refs/tags/}/CHANGELOG.md#${GITHUB_REF#refs/tags/} > ../release_notes.md 
+      - run: echo https://github.com/tendermint/tendermint/blob/${GITHUB_REF#refs/tags/}/CHANGELOG.md#${GITHUB_REF#refs/tags/} > ../release_notes.md
```
Comment (cmwaters) [inline .github/workflows/release.yml:23]: Do we still need to run this command?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -35,6 +35,6 @@ jobs:
         if: startsWith(github.ref, 'refs/tags/')
         with:
           version: latest
-          args: release --rm-dist --release-notes=../release_notes.md
+          args: release --rm-dist
```
Comment (marbar3778) [inline .github/workflows/release.yml:38]: just a note, this removes automatically adding messages to releases. After this point the releaser will have to manually edit the release.

File: .github/workflows/release.yml (hunk 1 above)
Comment (marbar3778) [reply .github/workflows/release.yml:23]: if you dont use it below then no

File: .github/workflows/release.yml (hunk 2 above)
Comment (cmwaters) [reply .github/workflows/release.yml:38]: What messages do we usually add to our releases?

Comment (marbar3778) [reply .github/workflows/release.yml:38]: links to the changelog https://github.com/tendermint/tendermint/blob/v0.34.12/CHANGELOG.md#v0.34.12.

the sdk just edits it, so we can do the same.

Comment (cmwaters) [reply .github/workflows/release.yml:38]: I don't mind manually doing it - it's not too much work

File: .github/workflows/release.yml (hunk 1 above)
Comment (cmwaters) [reply .github/workflows/release.yml:23]: Ok so we can remove this right?

Comment (marbar3778) [reply .github/workflows/release.yml:23]: yes

**Review by marbar3778 — APPROVED** [review]

**Pull request merged**